docs/developer_guide/contributing.rst

```diff
@@ -0,0 +1,420 @@
+                return_mapping=lambda x: 'T={},P={}'.format(x[0], x[1])),
+        }
+
+        in_terminator = '\r\n'
+        out_terminator = '\r\n'
+
+        def get_status(self):
+            return self._device.state
+
+        def get_position(self):
+            return self._device.position
+
+        def get_target(self):
+            return self._device.target
+
+        def set_target(self, new_target):
+            try:
+                self._device.target = new_target
+                return 'T={}'.format(new_target)
+            except RuntimeError:
+                return 'err: not idle'
+            except ValueError:
+                return 'err: not 0<=T<=250'
+
+The first argument to :class:`~plankton.adapters.stream.Cmd` specifies the method
+name the command is bound to, whereas the second argument is the regular expression that a
+request coming in over the TCP stream must match. If a method has
+arguments (such as ``set_target``), these need to be defined as capture
+groups in the regular expression. These groups are passed as strings to
+the bound method. If any sort of conversion is required for these
+arguments, the ``argument_mapping``-parameter can be a tuple of
+conversion functions with the same lengths as the number of capture
+groups in the regular expression. In the case of ``set_target`` it's
+enough to convert the string to float. Return values (except ``None``)
+are converted to strings automatically, but this conversion can be
+overridden by supplying a callable object to ``return_mapping``, as it
+is the case for the ``stop``-command.
+
+You may have noticed that ``stop`` is not a method of the interface.
+:class:`~plankton.adapters.stream.StreamAdapter` tries to resolve the supplied method
+names in multiple ways. First it checks its own members, then it checks the members of the
+device it owns (accessible in the interface via the ``_device``-member)
+and adds forwarders to itself if possible. If the method name can not be
+found in either the device or the adapter, an error is produced, which
+minimizes the likelihood of typos. The definitions in the interface
+always have precedence, this is intentionally done so that device
+behavior can be overridden later on with minimal changes to the code.
+
+In case of the ``stop``-method, which returns two floating point numbers
+(target and position), the ``return_mapping`` is used to format the
+device's position and target as specified in the protocol definition at
+the top of the page.
+
+Finally, in- and out-terminators need to be specified. These are
+stripped from and appended to requests and replies respectively.
+
+This entire device can be found in the ``examples`` directory. It can be
+started using the ``-k`` parameter of ``plankton.py``:
+
+::
+
+    $ ./plankton.py -k plankton.examples example_motor -- -b 127.0.0.1 -p 9999
+
+All functionality described in the
+:ref:`user_guide`, such as accessing the device and the simulation via the
```

Comment:
Here I think it's actually not so bad to use `:ref:`, because that makes linking more flexible (i.e. the document name doesn't need to be known).